apps/game_backend/lib/game_backend/users.ex

```diff
@@ -533,4 +573,132 @@ defmodule GameBackend.Users do
 
     Repo.all(q)
   end
+
+  defp add_quest_progress_and_goal(%User{} = user) do
+    updated_quests =
+      Enum.map(user.user_quests, fn user_quest ->
+        quest_progress =
+          Quests.get_user_quest_progress(user_quest, user.arena_match_results)
+
+        Map.put(user_quest, :progress, quest_progress)
+        |> Map.put(:goal, user_quest.quest.objective["value"])
+      end)
+
+    Map.put(user, :user_quests, updated_quests)
+  end
+
+  defp add_quest_progress_and_goal(_), do: nil
+
+  def insert_user_and_insert_daily_quests() do
+    curse_id = GameBackend.Utils.get_game_id(:curse_of_mirra)
+
+    Multi.new()
+    |> Multi.run(:insert_user, fn _, _ ->
+      CurseUsers.create_user_params()
+      |> register_user()
+    end)
+    |> Multi.run(:generate_quests, fn
+      _, %{insert_user: user} ->
+        quests_results = generate_daily_quests_for_user(user)
+
+        any_failed_quest? = Enum.find(quests_results, fn {result, _quest} -> result == :error end)
+
+        case any_failed_quest? do
+          {:error, changeset} ->
+            {:error, changeset}
+
+          _ ->
+            {:ok, :quests_generated}
+        end
+    end)
+    |> Multi.run(:user, fn _, %{insert_user: user} ->
+      get_user_by_id_and_game_id(user.id, curse_id)
+    end)
+    |> Repo.transaction()
+  end
+
+  def get_user_and_maybe_insert_daily_quests(user_id) do
+    curse_id = GameBackend.Utils.get_game_id(:curse_of_mirra)
+
+    Multi.new()
+    |> Multi.run(:get_user, fn _, _ ->
+      get_user_by_id_and_game_id(user_id, curse_id)
+    end)
+    |> Multi.run(:maybe_generate_quests, fn
+      _, %{get_user: user} ->
+        today = Date.utc_today()
+
+        should_generate_quests? =
+          user.last_daily_quest_generation_at &&
+            Date.compare(NaiveDateTime.to_date(user.last_daily_quest_generation_at), today) == :lt
+
+        if should_generate_quests? do
+          quests_results = generate_daily_quests_for_user(user)
+          any_failed_quest = Enum.find(quests_results, fn {result, _quest} -> result == :error end)
```

Comment:
Same here regarding the failed quest naming.

Reply:
I'll change the failed with an error